### ISSUE TITLE
vagrant uses wrong ip for ssh forward

### ISSUE THREAD

**Issue by asoki** [hashicorp/vagrant#10868]
### Vagrant version
Installed Version: 2.2.4
Latest Version: 2.2.4
vagrant-vmware-desktop (2.0.3, global)
vmware fusion Professional Version 11.1.0

### Host operating system
macOS high sierra 10.13.6 (17G5019) 

### Guest operating system
Devuan ascii 2.0.0

### Vagrantfile
```ruby
config.vm.define 'foodev94' do |foo_config|
  foo_config.vm.box = 'foo4'
  foo_config.vm.guest = :debian
  foo_config.vm.box_url = 'file:///Users/user1/path/to/box'
  foo_config.vm.host_name = 'foodev94'
  foo_config.vm.synced_folder '.', '/vagrant', disabled: true
  foo_config.vm.network "private_network", ip: "192.168.116.94"
  foo_config.vm.synced_folder "./","/foo-server/foo-code/foo-trunk", id:"foodev-code-sync", type:"nfs", mount_options: ["actimeo=2"]
  foo_config.vm.provision "shell", path: "vagrant_config/foodev_provisioner.sh"
  #foo_config.ssh.insert_key = false
  foo_config.ssh.private_key_path="~/.vagrant_pk/vagrant.foo.foobar.ch_id_rsa"
  foo_config.vm.provider "vmware_fusion" do |v|
    v.vmx['displayname'] = 'foodev94 Linux'
    v.vmx["memsize"] = "2048"
    v.vmx["numvcpus"] = "2"
    v.linked_clone = false
    v.gui = true
  end
end
```

### Debug output
https://gist.github.com/asoki/4e480eeab835711d7b535636f6744da0

### Expected behavior
vagrant should forward the ip belonging to the guest and should be able to log in via ssh
or
provide a config option to optionally tell vagrant the expected boot time of the guest and so allow to manually delay vagrants bootup processing

### Actual behavior
Vagrant requests the ip address of the guest from vmware and it gets the wrong ip.
See log at: DEBUG vagrant_utility: request METHOD=GET PATH=/vmnet/vmnet8/dhcplease/00:0c:29:06:4a:18 RESPONSE=200 
Vagrant gets 192.168.116.164 as the actual IP of the guest, but the currently booted guest got 192.168.116.165 from dhcp.
I think, this happens because vagrant asks too early for the ip, before the guest is fully booted.
The ip .164 did belonged to the previous try of the same box, i destroyed before.

At the next try to bring this box up vagrant will get the answer 192.168.116.165 as ip and the box .166. from dhcp, and so on.
The very first try to bring this box up failed because vagrant did not get any ip for this box.

### Steps to reproduce
1. create a minimal devuan-ascii-2.0.0 box (in tasksel: select only ssh and standard system utils)
2. try to boot it with vagrant

**Comment by chrisroberts** [comment]
Hi there,

Does your Vagrantbox include a Vagrantfile which explicitly defines a mac address for the guest?

**Comment by asoki** [comment]
as far as i know, it does not includes a Vagrantfile. i did not defined manually a mac address.
i created a new VM in VMWare Fusion, then installed the OS from official iso, added some custom apps to /usr/local, and with tar -cvfz created a box.
where should i look for that Vagrantfile?

**Comment by asoki** [comment]
i searched recursively for Vagrantfile on the whole disk of the guest, but found nothing.

**Comment by asoki** [comment]
i was able to workaround this issue with adding
`foo_config.vm.base_address = '192.168.116.93`
and configuring the network interface with the fixed 192.168.116.93 address

**Comment by briancain** [comment]
Hi there,

Thanks for reporting this bug. I'm sure this was a real issue when originally reported (our fault for not looking sooner!) but there have been multiple Vagrant releases since the original report. I'm going to close this issue now and request you reopen the issue if you're still experiencing this problem. I'm sorry this wasn't looked at earlier 

Cheers!
